lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb

```diff
@@ -110,34 +130,95 @@ def fetch_config
           break
         end
       end
-
       nil
     end
 
     def set_config(config, etag, lastmodified)
       if !JSON.parse(config).is_a?(Hash)
         raise("Invalid JSON body parsed from Config Response")
       end
-
+      parsed_config = JSON.parse(config)
+
+      if parsed_config['sse'] != nil
+        raw_url = "#{parsed_config['sse']['hostname']}#{parsed_config['sse']['path']}"
+        if @sse_url != raw_url && raw_url != ""
+          @sse_url = raw_url
+          stop_sse
+          init_sse(@sse_url)
+        end
+      end
       @local_bucketing.store_config(config)
       @config_e_tag = etag
       @config_last_modified = lastmodified
       @local_bucketing.has_config = true
+      @logger.debug("New config stored, etag: #{@config_e_tag}, last modified: #{lm_header}")
     end
 
     def get_config_url
       configBasePath = @local_bucketing.options.config_cdn_uri
       "#{configBasePath}/config/#{@config_version}/server/#{@sdkKey}.json"
     end
 
-    def stop_polling
+    def start_polling(sse)
+      if sse
+        @config_poller.shutdown if @config_poller.running?
+        @config_poller = Concurrent::TimerTask.new({ execution_interval: 60 *10 }) do |_|
+          fetch_config
+        end
+      end
+      @polling_enabled = true
+      @config_poller.execute if @polling_enabled && (!@sse_active || sse)
+    end
+
+    def stop_polling()
       @polling_enabled = false
       @config_poller.shutdown if @config_poller.running?
     end
 
+    def stop_sse
+      return unless @enable_sse
+      @polling_enabled = true
+      @sse_active = false
+      @sse_client.close if @sse_client
+      start_polling(false)
+    end
+
     def close
       @config_poller.shutdown if @config_poller.running?
       nil
     end
+
+    def init_sse(path)
+      return unless @enable_sse
+      @logger.debug("Initializing SSE with url: #{path}")
+      @sse_active = true
+      @sse_client = SSE::Client.new(path) do |client|
+        client.on_event do |event|
+          parsed_json = JSON.parse(event.data)
+          handle_sse(parsed_json)
+        end
+        client.on_error do |error|
+          @logger.debug("SSE Error: #{error.message}")
+        end
+      end
+      stop_polling
+      start_polling(true)
+    end
+
+    def handle_sse(eventData)
+      if eventData["data"] == nil
+        return
+      end
+      @logger.debug("SSE Message received: #{eventData["data"]}")
+      parsed_event_data = JSON.parse(eventData["data"])
+
+      last_modified = parsed_event_data["lastModified"]
+      event_type = parsed_event_data["type"]
+
+      if event_type == "refetchConfig" || event_type == nil
+        @logger.debug("Re-fetching new config with TS: #{last_modified}")
+        fetch_config(min_last_modified: last_modified / 1000)
```

Comment:
will this work if last_modified isnt set

Reply:
yes - it'll result in the arg being `0` - which will just request a new config